apps/server/src/services/rundown-service/delayedRundown.utils.ts

```diff
@@ -116,6 +116,12 @@ export async function cachedEdit(
   return newEvent;
 }
 
+export async function cachedBatchEdit(ids: string[], patchObject: Partial<OntimeEvent>) {
+  const cachedEdits = ids.map((id) => cachedEdit(id, patchObject));
+
+  await Promise.all(cachedEdits);
+}
+
```

Comment:
This works well, but probably not too computationally efficient. If you think users would be editing 100s of events simultaneously, then I'd need to re-write this logic;

Event if one `cachedEdit` fails (wrong id etc), using `Promise.all` would fail _all_ edits. This is intentional because I thought it might confuse users to see all except one succeed. I could be wrong with my assumptions.

Reply:
hmm, thinking out loud:
using `promise.all` we reject if any of the promises reject, this means;
1. the events that were already edited will remain edited
2. we will not attempt editing any further on a fail
3. the user will receive a promise rejection on the request, but see some changes

Without a way to roll back the edits, would it be best to use `Promise.settled` to continue through the list of edits? Is there an error code that describes "we made some changes, some stuff failed"?

What do you think? Are the above assumptions correct?